### ISSUE TITLE
Evaluate RAM memory encryption solution to improve resiliency against memory dump

### ISSUE THREAD

**Issue by fpietrosanti** [globaleaks/GlobaLeaks#685]
This ticket is to evaluate RAM memory encryption technical solution to improve resiliency against memory dump.

The core security improvement would be to defeat simpler memory dump with string analysis done against a virtual machine or against a physical machine with direct memory access.

The system must not be "perfect" but should introduce resiliency against basic attacks.

The implementation have to be lightweight and with no major compatibility issue.

**Comment by origliante** [comment]
@fpietrosanti partially outdated by end2end?

**Comment by fpietrosanti** [comment]
@origliante Why is outdated?

**Comment by hellais** [comment]
Yeah mlock is the correct solution to this matter, but it would require us to write a special python C based binding.

**Comment by evilaliv3** [comment]
The RAM encryption is Implemented since time; the only missing part is the mlockall(1|2) that is now tracked by a specific ticket.

**Comment by fpietrosanti** [comment]
@evilaliv3 If it's implemented (are we sure?!?), it must be documented in the Application Technical Security Document to close this ticket, explaining how does it works. Afaik SD did managed to do it only with custom C-code patching and kernel's PAX feature that we are not using

**Comment by evilaliv3** [comment]
ah no you are right, i was confusing this ticket with the encryption in RAM.
